pkg/platform/kube/monitoring/test/function_test.go

```diff
@@ -0,0 +1,312 @@
+package test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/common"
+	"github.com/nuclio/nuclio/pkg/functionconfig"
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube"
+	"github.com/nuclio/nuclio/pkg/platform/kube/monitoring"
+	kubetest "github.com/nuclio/nuclio/pkg/platform/kube/test"
+
+	"github.com/stretchr/testify/suite"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type FunctionMonitoringTestSuite struct {
+	kubetest.KubeTestSuite
+}
+
+func (suite *FunctionMonitoringTestSuite) SetupSuite() {
+	suite.KubeTestSuite.SetupSuite()
+
+	// decrease blocking interval, to make test run faster
+	monitoring.PostDeploymentMonitoringBlockingInterval = time.Second
+}
+
+func (suite *FunctionMonitoringTestSuite) TestNoRecoveryAfterBuildError() {
+	functionName := "function-build-fail"
+	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
+	getFunctionOptions := &platform.GetFunctionsOptions{
+		Name:      createFunctionOptions.FunctionConfig.Meta.Name,
+		Namespace: createFunctionOptions.FunctionConfig.Meta.Namespace,
+	}
+
+	// fail during build
+	createFunctionOptions.FunctionConfig.Spec.Build.Commands = []string{
+		"exit 1",
+	}
+
+	// wait for at least one monitor + post deployment blocking intervals
+	waitOnMonitoringIntervals := 2*suite.Controller.GetFunctionMonitoringInterval() +
```

Comment:
```suggestion
	postDeploymentSleepInterval := 2*suite.Controller.GetFunctionMonitoringInterval() +
```
Or something else clear (describe the interval not "waitOn" action which is action induced by using this in `time.Sleep()`)